code/controllers/subsystem/xenoarchaeology.dm

```diff
@@ -0,0 +1,215 @@
+SUBSYSTEM_DEF(xenoarchaeology)
+	name = "Xenoarchaeology"
+	flags = SS_NO_FIRE
+	init_order = INIT_ORDER_XENOARCHAEOLOGY
+
+	///Which console is the main character
+	var/obj/machinery/computer/xenoarchaeology_console/main_console
+
+	///Traits types, referenced for generation
+	var/list/xenoa_activators
+	var/list/xenoa_minors
+	var/list/xenoa_majors
+	var/list/xenoa_malfunctions
+	var/list/xenoa_all_traits
+	///All traits indexed by name
+	var/list/xenoa_all_traits_keyed
+
+	///Names for science sellers & artifacts
+	var/list/xenoa_seller_names
+	var/list/xenoa_seller_dialogue
+	var/list/xenoa_artifact_names
+
+	///Whitelist for traits by type
+	var/list/xenoa_bluespace_traits
+	var/list/xenoa_plasma_traits
+	var/list/xenoa_uranium_traits
+	var/list/xenoa_bananium_traits
+	var/list/xenoa_pearl_traits
+
+	///Incompatability lists - Partial future proof
+	var/list/xenoa_item_incompatible
+	var/list/xenoa_mob_incompatible
+	var/list/xenoa_structure_incompatible
+
+	///Labeler trait lists
+	var/list/labeler_activator_traits
+	var/list/labeler_minor_traits
+	var/list/labeler_major_traits
+	var/list/labeler_malfunction_traits
+	var/list/labeler_tooltip_stats = list()
+	var/list/labeler_traits_filter
+
+	///Material weights, basically rarity
+	var/list/xenoartifact_material_weights = list(XENOA_BLUESPACE = 10, XENOA_PLASMA = 8, XENOA_URANIUM = 5, XENOA_BANANIUM = 1)
+
+	///Trait priority list - The order is important and it represents priotity
+	var/list/xenoartifact_trait_priorities = list(TRAIT_PRIORITY_ACTIVATOR, TRAIT_PRIORITY_MINOR, TRAIT_PRIORITY_MALFUNCTION, TRAIT_PRIORITY_MAJOR)
+
+	///List of 'discovered' traits
+	var/list/discovered_traits = list()
+
+/datum/controller/subsystem/xenoarchaeology/Initialize(timeofday)
+	. = ..()
+	//Poplate seller & artifact personalities
+	xenoa_seller_names = world.file2list("strings/names/science_seller.txt")
+	xenoa_seller_dialogue = world.file2list("strings/science_dialogue.txt")
+	xenoa_artifact_names = world.file2list("strings/names/artifact_sentience.txt")
+	//Bruh
+	xenoa_seller_names -= ""
+	xenoa_seller_dialogue -= ""
+	xenoa_artifact_names -= ""
```

Comment:
If this is needed then there is an issue with file2list that needs fixing instead. Other things depend on this not containing an empty line at the end, so it shouldnt.